src/mapgen_v7.cpp

```diff
@@ -76,9 +79,14 @@ MapgenV7::MapgenV7(int mapgenid, MapgenV7Params *params, EmergeManager *emerge)
 	// 1-up 1-down overgeneration
 	noise_mountain = new Noise(&params->np_mountain, seed, csize.X, csize.Y + 2, csize.Z);
 	noise_ridge    = new Noise(&params->np_ridge,    seed, csize.X, csize.Y + 2, csize.Z);
+	// 1 down overgeneration
+	noise_cavern   = new Noise(&params->np_cavern,   seed, csize.X, csize.Y + 1, csize.Z);
```

Comment:
It needs to be a separate commit, and i am not yet comfortable with how to edit an earlier commit once there's a second commit. I think i'd prefer a separate PR then i can do the others at the same time.